pkg/scheduler/preemption/preemption.go

```diff
@@ -168,6 +179,21 @@ func minimalPreemptions(wl *workload.Info, assignment flavorassigner.Assignment,
 		if cq != candCQ && !cqIsBorrowing(candCQ, resPerFlv) {
 			continue
 		}
+		if cq != candCQ && allowBorrowingBelowPriority != nil && priority.Priority(candWl.Obj) >= *allowBorrowingBelowPriority {
+			// We set allowBorrowing=false if there is a candidate with priority
+			// exceeding allowBorrowingBelowPriority added to targets.
+			//
+			// We need to be careful mutating allowBorrowing. We rely on the
+			// fact that once there is a candidate exceeding the priority added
+			// to targets, then at least one such candidate is present in the
+			// final set of targets (after the second phase of the function).
+			//
+			// This is true, because the candidates are ordered according
+			// to priorities (from lowest to highest, using candidatesOrdering),
+			// and the last added target is not removed in the second phase of
+			// the function.
+			allowBorrowing = false
```

Comment:
I'm no longer sure if this works.

What if, in a previous iteration, we were already borrowing? Then `workloadFits` would return false and break this loop. So far so good.

However, the next loop (check if any of the workloads can be added back) would be prevented from borrowing, even though it was ok to borrow. As a result, we won't be able to add back any workload, possibly leading to preempting more workloads than necessary.

Reply:
> What if, in a previous iteration, we were already borrowing? Then workloadFits would return false and break this loop. So far so good.

The `workloadFits` breaks when returning `true` (not false), snippet from below:

```golang
		if workloadFits(wlReq, cq, allowBorrowing) {
			fits = true
			break
		}
```

Reply:
ohhhh, right.

So the mechanics roughly like this:
1. allowBorrowing=true, workload doesn't fit,
2. we pass the threshold, workload still doesn't fit, we can continue deleting workloads, but now we cannot allow borrowing.
3. Eventually, the workload fits. We are already deleting workloads past the threshold, so if we put workloads back, we cannot allow borrowing anyway.

Ack.